flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -453,9 +457,10 @@ public void testRequestBuffersWithUnknownInputChannel() throws Exception {
 			}
 
 			// Trigger updates to remote input channel from unknown input channel
-			inputGate.updateInputChannel(new InputChannelDeploymentDescriptor(
-				resultPartitionId,
-				ResultPartitionLocation.createRemote(InputChannelBuilder.STUB_CONNECTION_ID)));
+			ShuffleDescriptor sdd = createRemoteWithIdAndLocation(
+				resultPartitionId.getPartitionId(),
+				ResourceID.generate());
+			inputGate.updateInputChannel(ResourceID.generate(), new PartitionInfo(new IntermediateDataSetID(), sdd));
```

Comment:
could reuse createPartitionInfo(resultPartitionId, ResourceID.generate()) to create PartitionInfo here.